src/user-event/press/press.ts

```diff
@@ -1,16 +1,118 @@
 import { ReactTestInstance } from 'react-test-renderer';
 import { EventBuilder } from '../event-builder';
 import { UserEventInstance } from '../setup';
-import { dispatchHostEvent, wait } from '../utils';
+import { wait } from '../utils';
+import act from '../../act';
+import { getHostParent } from '../../helpers/component-tree';
+import { filterNodeByType } from '../../helpers/filterNodeByType';
+import { isPointerEventEnabled } from '../../helpers/pointer-events';
+import { getHostComponentNames } from '../../helpers/host-component-names';
+import { jestFakeTimersAreEnabled } from '../../helpers/timers';
+import { DEFAULT_MIN_PRESS_DURATION } from './constants';
+import { warnAboutRealTimers } from './utils/warnAboutRealTimers';
+
+export type PressOptions = {
+  duration: number;
+};
 
 export async function press(
   this: UserEventInstance,
   element: ReactTestInstance
-) {
-  // TODO provide real implementation
-  dispatchHostEvent(element, 'pressIn', EventBuilder.Common.touch());
+): Promise<void> {
+  await basePress(this.config, element);
+}
 
-  await wait(this.config);
-  dispatchHostEvent(element, 'press', EventBuilder.Common.touch());
-  dispatchHostEvent(element, 'pressOut', EventBuilder.Common.touch());
+export async function longPress(
+  this: UserEventInstance,
+  element: ReactTestInstance,
+  options: PressOptions = { duration: 500 }
+): Promise<void> {
+  await basePress(this.config, element, options);
 }
+
+const basePress = async (
+  config: UserEventInstance['config'],
+  element: ReactTestInstance,
+  options: PressOptions = { duration: 0 }
+): Promise<void> => {
+  // Text and TextInput components are mocked in React Native preset so the mock
+  // doesn't implement the pressability class
+  // Thus we need to call the props directly on the host component
+  const isEnabledHostText =
+    filterNodeByType(element, getHostComponentNames().text) &&
+    isPointerEventEnabled(element) &&
+    !element.props.disabled;
+  const isEnabledTextInput =
+    filterNodeByType(element, getHostComponentNames().textInput) &&
+    isPointerEventEnabled(element) &&
+    element.props.editable !== false;
+
+  if (isEnabledHostText || isEnabledTextInput) {
+    const { onPressIn, onPress, onPressOut } = element.props;
+    await wait(config);
+    if (onPressIn) {
+      onPressIn(EventBuilder.Common.press());
+    }
+    if (onPress) {
+      onPress(EventBuilder.Common.press());
+    }
+    await wait(config, options.duration);
+    if (onPressOut) {
+      if (DEFAULT_MIN_PRESS_DURATION - options.duration > 0) {
```

Comment:
Is that 130 ms happening experimentally in an actual RN app for `Text`/`TextInput`? We should consider adding it artificially if this is the case.

Reply:
Yes it happens also for `Text` and `TextInput`because they use the same implementation of `Pressability`. Not sure I understand your comment though, we're already adding it artificially here